content/200-concepts/100-components/02-prisma-client/053-client-extensions/150-type-utilities.mdx

```diff
@@ -23,50 +23,81 @@ The type utilities available are:
+The following example creates a new operation, `exists`, based on `findFirst`. It has all of the arguments that `findFirst`.
 
 ```ts
-const prisma = new PrismaClient()
+const prisma = new PrismaClient().$extends({
+  model: {
+    $allModels: {
+      // Define a new `exists` operation on all models
+      // T is a generic type that corresponds to the current model
+      async exists<T>(
+        // `this` refers to the current type, e.g. `prisma.user` at runtime
+        this: T,
+
+        // The `exists` function will use the `where` arguments from the current model, `T`, and the `findFirst` operation
+        where: Prisma.Args<T, 'findFirst'>['where']
+      ): Promise<boolean> {
+        // Retrieve the current model at runtime
+        const context = Prisma.getExtensionContext(this)
 
-const xprisma = prisma.$extends({
+        // Prisma Client query that retrieves data based
+        const result = await (context as any).findFirst({ where })
+        return result !== null
+      },
+    },
+  },
+})
+
+async function main() {
+  const user = await prisma.user.exists({ name: 'Alice' })
+  const post = await prisma.post.exists({
+    OR: [
+      { title: { contains: 'Prisma' } },
+      { content: { contains: 'Prisma' } },
+    ],
+  })
+}
+```
+
+## Add a custom property to a method
+
+The following example illustrates how you can add custom arguments, to a method in an extension:
+
+```ts highlight=17
+type CacheStrategy = {
+  swr: number
+  ttl: number
+}
+
+const prisma = new PrismaClient().$extends({
   model: {
     $allModels: {
-      // Define a new operation `customCall`.
-      // T corresponds to the current model,
-      // A corresponds to the arguments for the operation.
-      customCall<T, A>(
-        // `this` is the current type (for example
-        // it might be `prisma.user` at runtime).
+      // `T` is a generic type that corresponds to the current model
+      // `A` is a generic type that corresponds to the argument for the operation
+      findMany<T, A>(
         this: T,
-        x: Prisma.Exact<
+        args: Prisma.Exact<
           A,
-          // For `customCall`, use the arguments from model `T` and the
-          // operation `findFirst`. Add `customProperty` to the operation.
-          Prisma.Args<T, 'findFirst'> & { customProperty: boolean }
+          // For the `findMany` method, use the arguments from model `T` and the `findMany` method
+          // and intersect it with `CacheStrategy` as part of `findMany` arguments
+          Prisma.Args<T, 'findMany'> & CacheStrategy
         >
-
-        // Get the correct result types for the model of type `T`,
-        // and the arguments of type `A` for `findFirst`.
-        // `Prisma.Result` computes the result for a given operation
-        // such as `select {id: true}` in function `main` below.
-        //
-      ): Prisma.Result<T, A, 'findFirst'> {
```

Comment:
Does `Prisma.Result` have a spot where it is showcased?

Reply:
I've added it to the caching example.